specification

hettlage opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/saltastro/token-auth-request/1)
<!-- Reviewable:end -->

hettlage committed:
specification

hettlage committed:
updated test description

hettlage committed:
initial tests

hettlage committed:
ignore Python caches

hettlage committed:
ignore .hypothesis folders

hettlage committed:
updated specification

hettlage committed:
implementation and tests

hettlage committed:
added documentation

hettlage committed:
removed obsolete constant

hettlage committed:
support Python 2.7 and 3.6

hettlage committed:
updated testing configuration

hettlage committed:
use authentication with custom HTTP verbs

hettlage committed:
test authentication with all HTTP verbs

hettlage committed:
renamed package

hettlage committed:
dropped Python 2.7 support

hettlage committed:
updated specification

hettlage committed:
test for auth_request_maker

hettlage committed:
implementation of auth_request_maker

chynamyerz commented:
I went through the specs quickly and they do make sense. Please see the comments I made. Other than that, fine by me.

One thing though I found challenging is the fact that I am not familiar with this kind of testing which caused to take more time trying to follow. I don't know if it will be possible we all look at it while you taking us through it explaining. :-) If that it okay with everyone.

---

Review status: 0 of 14 files reviewed at latest revision, all discussions resolved.

---

*[token_auth_requests.py, line 33 at r1](https://reviewable.io/reviews/saltastro/token-auth-request/1#-LBZq0G_-yeU0fvEyllG:-LBZq0Ga9L3x12LcaCUX:bop4lub) ([raw file](https://github.com/saltastro/token-auth-request/blob/1e49be7e7468438152d65792eb92294243f3e931/token_auth_requests.py#L33)):*
> ```Python
>     it gets an authentication token by sending the username and password to an authentication URL. The token and its
>     expiry date are stored, and the token is sent in an Authentication HTTP header with all HTTP requests. If the
>     token expires, a new one is requested before the next HTTO request is made.
> ```

HTTO => HTTP

---

*[docs/quickstart.rst, line 40 at r1](https://reviewable.io/reviews/saltastro/token-auth-request/1#-LBWGv_WDpjoKvj2bNuP:-LBWGv_WDpjoKvj2bNuQ:b4jec68) ([raw file](https://github.com/saltastro/token-auth-request/blob/1e49be7e7468438152d65792eb92294243f3e931/docs/quickstart.rst#L40)):*
> ```reStructuredText
>    print(r.status_code)  # hopefully the status code is 200...
> 
> Refer to the `requests documentation <http://docs.python-requests.org/>`_ for all the cool stuff you can do. You don't have to worry about about tokens or their expiry dates; trhey are taken care of automatically.
> ```

trhey => they

---

*[specs/index.rst, line 58 at r1](https://reviewable.io/reviews/saltastro/token-auth-request/1#-LBWI-l84MiIFEtsrJsu:-LBWI-l84MiIFEtsrJsv:b-15y41k) ([raw file](https://github.com/saltastro/token-auth-request/blob/1e49be7e7468438152d65792eb92294243f3e931/specs/index.rst#L58)):*
> ```reStructuredText
> 
> 
> Both the token and its expiry time are stored. After that, or if there was a non-expired token already, the token is added as an HTTP header with key `Authentication` and value `Token xyz` (where `xyx` denotes the token) to the session. For example, for the token above the header would look be
> ```

xyx => xyz

---


*Comments from [Reviewable](https://reviewable.io/reviews/saltastro/token-auth-request/1#-:-LBWJ4OgBGlvzDcEAoOj:bu89wor)*
<!-- Sent from Reviewable.io -->

hettlage committed:
Fixed typos

hettlage committed:
test and implementation of auth_response_parser

hettlage committed:
updated documentation

hettlage commented:
Review status: 0 of 14 files reviewed at latest revision, 3 unresolved discussions.

---

*[token_auth_requests.py, line 33 at r1](https://reviewable.io/reviews/saltastro/token-auth-request/1#-LBZq0G_-yeU0fvEyllG:-LBc-8qrCl_xHEDicPpG:bt2vr1b) ([raw file](https://github.com/saltastro/token-auth-request/blob/1e49be7e7468438152d65792eb92294243f3e931/token_auth_requests.py#L33)):*
<details><summary><i>Previously, chynamyerz wrote…</i></summary><blockquote>

HTTO => HTTP
</blockquote></details>

Fixed.

---

*[docs/quickstart.rst, line 40 at r1](https://reviewable.io/reviews/saltastro/token-auth-request/1#-LBWGv_WDpjoKvj2bNuP:-LBc-CttAL7sU6DmHktq:bt2vr1b) ([raw file](https://github.com/saltastro/token-auth-request/blob/1e49be7e7468438152d65792eb92294243f3e931/docs/quickstart.rst#L40)):*
<details><summary><i>Previously, chynamyerz wrote…</i></summary><blockquote>

trhey => they
</blockquote></details>

Fixed.

---

*[specs/index.rst, line 58 at r1](https://reviewable.io/reviews/saltastro/token-auth-request/1#-LBWI-l84MiIFEtsrJsu:-LBc-Fmj1DlNfvWD7mvg:bt2vr1b) ([raw file](https://github.com/saltastro/token-auth-request/blob/1e49be7e7468438152d65792eb92294243f3e931/specs/index.rst#L58)):*
<details><summary><i>Previously, chynamyerz wrote…</i></summary><blockquote>

xyx => xyz
</blockquote></details>

Fixed.

---


*Comments from [Reviewable](https://reviewable.io/reviews/saltastro/token-auth-request/1)*
<!-- Sent from Reviewable.io -->

chynamyerz commented:
Reviewed 9 of 14 files at r1, 5 of 5 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io/reviews/saltastro/token-auth-request/1)*
<!-- Sent from Reviewable.io -->
